.github/workflows/api-checks.yml

```diff
@@ -0,0 +1,55 @@
+name: Template API Checks
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - app/**
+      - .github/workflows/api-checks.yml
+  pull_request:
+    paths:
+      - app/**
+      - .github/workflows/api-checks.yml
+
+defaults:
+  run:
+    working-directory: ./app
+
+env:
+  # Some commands in the Makefile check if this is set
+  # and add additional parameters to the commands called
+  CI: true
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: build container
+        run: make build
+
+      - name: Run format check
+        run: make format-check
+
+      - name: Run linting
+        run: make lint
+  security:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Security Check
+        run: make lint-security
+  test:
+    needs: [lint, security]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: test for multi-headed migration situation
+        run: make test args="-x tests/api/db/test_migrations.py::test_only_single_head_revision_in_migrations" 
+
+      - name: Start tests
+        run: make test-coverage
```

Comment:
nit: missing newline at end of file

Reply:
Should we add a tool for linting/formatting of yaml files to the backlog?

Reply:
👍 , created #25